main.go

```diff
@@ -5,30 +5,72 @@ package main
 
 import (
 	"crypto/ed25519"
+	_ "embed"
+	"errors"
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"os/signal"
 	"strings"
 	"syscall"
 
 	"github.com/spf13/pflag"
-	"github.com/tillitis/tkey-sign/internal/util"
 	"github.com/tillitis/tkeyclient"
 	"github.com/tillitis/tkeysign"
 )
 
+// nolint:typecheck // Avoid lint error when the embedding file is missing.
+// Build copies the built signer here
+//
+//go:embed signer.bin
+var signerBinary []byte
+
 // Use when printing err/diag msgs
 var le = log.New(os.Stderr, "", 0)
 
 // May be set to non-empty at build time to indicate that the signer
 // app has been compiled with touch requirement removed.
 var signerAppNoTouch string
 
+const (
+	// 4 chars each.
+	wantFWName0  = "tk1 "
+	wantFWName1  = "mkdf"
+	wantAppName0 = "tk1 "
+	wantAppName1 = "sign"
+)
+
+func isFirmwareMode(tk *tkeyclient.TillitisKey) bool {
+	nameVer, err := tk.GetNameVersion()
+	if err != nil {
+		return false
+	}
+	// not caring about nameVer.Version
+	return nameVer.Name0 == wantFWName0 &&
+		nameVer.Name1 == wantFWName1
+}
+
+func isWantedApp(signer tkeysign.Signer) bool {
+	nameVer, err := signer.GetAppNameVersion()
+	if err != nil {
+		if !errors.Is(err, io.EOF) {
+			le.Printf("GetAppNameVersion: %s\n", err)
+		}
+		return false
+	}
+
+	fmt.Printf("name0: %v, name1: %v\n", nameVer.Name0, nameVer.Name1)
+
+	// not caring about nameVer.Version
+	return nameVer.Name0 == wantAppName0 &&
+		nameVer.Name1 == wantAppName1
+}
+
 func main() {
 	var fileName, devPath string
 	var speed int
-	var showPubkeyOnly, verbose, helpOnly bool
+	var enterUSS, showPubkeyOnly, verbose, helpOnly bool
```

Comment:
Includes option for --uss but does not do anything.
Included for later development or missed?

Should there be a "not implemented" printout or something?
Since one could expect a prompt asking for a uss if one uses the flag.

Reply:
Implemented USS input and USS file.